tools/deploy.go

```diff
@@ -0,0 +1,736 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// GetEnvOrDefault returns the value of the environment variable key if it exists
+// and is non-empty. Otherwise it returns the provided default value.
+func GetEnvOrDefault(key, defaultValue string) string {
```

Comment:
I'd love if we could get rid of using env vars, or at least have an option to pass a config file directly to this script. For traditional reason we've always been tied up to env vars, and they have been very difficult to track, especially for newbies.

Of course changing it here doesn't mean we could get rid of the env vars, but it could be the the first step. In dev-env, for now we can add an ansible job to generate the config file based on the env vars we have, before running this script. For general users, it'd make more senses to ask them creating their config file than to populate all the env vars.

Reply:
Good idea, I will see if I find time to look into this.
@lentzi90 do you agree on doing this?

Reply:
Thanks Max. For the record, it's pretty fine for me if you don't implement this in the current PR. It looks like an improvement that deserves its own scope and user story.